src/re/main.c

```diff
@@ -476,6 +528,18 @@ endleaf_json(FILE *f, const void *state_opaque, const void *endleaf_opaque)
 	return 0;
 }
 
+static struct fsm *fsm_to_cleanup = NULL;
+
+static void
+do_fsm_cleanup(void)
+{
+	if (fsm_to_cleanup != NULL) {
+		fsm_free(fsm_to_cleanup);
```

Comment:
Explicit NULL; avoid use-after-free potential; even as this is the last code to run ;-) It's just cleaner to have it consistent as this symbol is globally visible.